lib/Service/AppsService.php

```diff
@@ -24,24 +24,51 @@ public function __construct(
 	public function findSupported(): array {
 		$apis = [];
 
+		$always = $this->appManager->getAlwaysEnabledApps();
 		foreach ($this->appManager->getInstalledApps() as $app) {
+			if (!$this->appManager->isEnabledForUser($app)) {
+				continue;
+			}
+			$appApis = [];
 			try {
 				$baseDir = $this->appManager->getAppPath($app);
 				$iterator = new \DirectoryIterator($baseDir);
 				foreach ($iterator as $file) {
 					if ($file->getFilename() == 'openapi.json') {
-						$apis[] = $app;
+						$appApis[] = $app;
 					} else if (str_starts_with($file->getFilename(), 'openapi-') && str_ends_with($file->getFilename(), '.json')) {
-						$apis[] = $app . '-' . substr($file->getFilename(), 8, -5);
+						$appApis[] = $app . '-' . substr($file->getFilename(), 8, -5);
 					}
 				}
 			} catch (AppPathNotFoundException) {
 			}
+			if (!empty($appApis)) {
+				$appInfo = $this->appManager->getAppInfo($app);
+				$apiInfo =[
+					'id' => $app,
+					'apis' => $appApis,
+					'name' => $appInfo['name'],
+					'version' => $appInfo['version'],
+					'standard' => in_array($app, $always),
+				];
+				$preview = $this->getPreview($app, $baseDir);
+				if ($preview !== null) {
+					$apiInfo['preview'] = $preview;
+				}
+				$apis[] = $apiInfo;
+			}
 		}
 
-		sort($apis);
-
-		array_unshift($apis, 'core');
+		if (file_exists(\OC::$SERVERROOT . '/core/openapi.json')) {
+			array_unshift($apis, [
+				'id' => 'core',
+				'apis' => ['core'],
+				'name' => 'Core Nextcloud API',
+				'version' => \OC_Util::getVersionString(),
```

Comment:
```suggestion
				'version' => implode('.', \OCP\Util::getVersion()),
```
This will be removed in 31 (see https://github.com/nextcloud/server/pull/48152) and lead to an internal server error on my development system.
Ideally we'd use the new Version API, but it is only available in 31.